pkg/analysis/helpers/inspector/inspector.go

```diff
@@ -0,0 +1,87 @@
+package inspector
+
+import (
+	"go/ast"
+	"go/token"
+
+	"github.com/JoelSpeed/kal/pkg/analysis/helpers/extractjsontags"
+	"github.com/JoelSpeed/kal/pkg/analysis/helpers/markers"
+	astinspector "golang.org/x/tools/go/ast/inspector"
+)
+
+// Inspector is an interface that allows for the inspection of fields in structs.
+type Inspector interface {
+	// InspectFields is a function that iterates over fields in structs.
+	InspectFields(func(field *ast.Field, stack []ast.Node, jsonTagInfo extractjsontags.FieldTagInfo, markersAccess markers.Markers))
```

Comment:
Nit: Make the input func a specific type?

Reply:
I had that, and moved away from it as my code completion was being annoying and not understanding how to populate... Maybe will move back in the future